gtests/test_nopython_io.cpp

```diff
@@ -0,0 +1,350 @@
+#include <modmesh/io/io.hpp>
```

Comment:
Too long to review.  I trust you have made comprehensive testing.  If not, please speak up.

Reply:
I have tested each cell type conversion between `gmsh` and `modmesh`. Additionally, I also tested if a user inputs a cell type that is not listed in `GmshElementDef`. The expected behavior is that `GmshElement::by_id` should return an empty `GmshElementDef` object.